content/2022/estimacija-pokreta.md

```diff
@@ -2,3 +2,133 @@
+
+### Random Field Model
+
+Za par predloga $l_{\mathbf{p}}$ i $l_{\mathbf{q}}$ susednih piksela $p$ i $q$ definišemo *cenu slaganja* $\psi_{\mathbf{p}, \mathbf{q}}\left(l_{\mathbf{p}}, l_{\mathbf{q}}\right)$ kao Menhetn normu njihove razlike.
+
+
+Kao i cena vektora protoka, odsečena je odozgo sa $\tau_\psi$.
+
+Polje protoka je modelirano kao Markovljev slučajni proces. Problem procene pokreta je sveden na minimizaciju zbira cena vektora protoka i cena slaganja.
+
+$$
+\lambda \sum_{\mathbf{p}} \varphi_{\mathbf{p}}\left(l_{\mathbf{p}}\right)+\sum_{\mathbf{p} \sim \mathbf{q}} \psi_{\mathbf{p}, \mathbf{q}}\left(l_{\mathbf{p}}, l_{\mathbf{q}}\right)
+$$
+
+Vrednost $\lambda$ je težinski faktor i u programu je jednak 0.05. Simbolima $p$ i $q$ su označeni pikseli na slici, susedni u desnoj sumi.
+
+### Pronalaženje optimalnih vektora
+
+U svakom trenutku se čuva trenutno polje protoka, i ukupna cena se postepeno smanjuje odabirom bolje kombinacije vektora protoka iz skupa vektora svakog piksela. To se radi pomoću spuštanja po blokovima koordinata (eng: *block coordinate descent), nalaženjem minimalnog zbira cena u pojedinačnom redu ili koloni piksela na slici. Ovo ne garantuje nalaženje globalnog minimuma cene, što je NP-težak problem, ali garantuje smanjivanje ukupne cene slike u svakom koraku spuštanja.
+
+U pojedinačnom redu piksela je potrebno naći kombinaciju vektora protoka iz predloga koja će dati minimalan zbir cena vektora protoka i cena slaganja, uzimajući u obzir trenutno stanje polja protoka u susedna dva reda. Cene na koje treba paziti su cene vektora protoka, cena slaganja sa pikselima iz susednih redova i cena slaganja sa pikselima iz istog reda.
+
+Problem minimizacije je rešen dinamičkom matricom $C$, čije su koordinate pozicija piksela u redu i redni broj trenutnog predloga u tom pikselu. Matrica se popunjava po jednačini:
+
+$$
+\begin{aligned}
+\mathbf{C}(x, l) &=\lambda \varphi_{(x, y)}(l)+\psi_{(x, y),(x, y-1)}\left(l, l_{x, y-1}^*\right)+\psi_{(x, y),(x, y+1)}\left(l, l_{x, y+1}^*\right) \\
+&+\min _{k \in L(x)}\left(\psi_{(x, y),(x-1, y)}(l, k)+\mathbf{C}(x-1, k)\right)
+\end{aligned}
+$$
+
+Algoritam kreće od početka reda piksela, i u $\mathbf{C}(x, l)$ čuva najmanju od kumulativnih cena svih kombinacija u delu reda od početnog do piksela rednog broja $x$, koje se završavaju predlogom $l$ tog piksela. Predlozi označeni $l_{x, y}^*$ su trenutni predlozi iz susednih redova na lokaciji $(x,y)$, a $L(x)$ skup predloga piksela $x$. Sami vektori protoka koji sačinjavaju tu kombinaciju se ne čuvaju, već se najbolja kombinacija rekonstruiše idući unazad kroz matricu nakon što se popuni do kraja. Stari predlozi vektora se zamene novim i kreće se sa obradom sledećeg reda.
+
+Drugi sabirak se ne računa grubom silom, već pomoću *k-skupova*.
+
+### K-skupovi
```

Comment:
Pošto se k-skupovi koriste prilikom pronalaženja optimalnih vektora, mislim da ovde fali još jedan #.